hyper/http20/response.py

```diff
@@ -71,10 +77,10 @@ def __init__(self, headers, stream):
         # This 16 + MAX_WBITS nonsense is to force gzip. See this
         # Stack Overflow answer for more:
         # http://stackoverflow.com/a/2695466/1401686
-        if b'gzip' in self.headers.get(b'content-encoding', []):
-            self._decompressobj = zlib.decompressobj(16 + zlib.MAX_WBITS)
-        elif b'deflate' in self.headers.get(b'content-encoding', []):
-            self._decompressobj = DeflateDecoder()
+        compressionTypes = set(self.headers.get(b'content-encoding', []))
+        compressionTypes &= set(decompressors.keys())
+        if compressionTypes:
+            self._decompressobj = decompressors[next(iter(compressionTypes))]()
```

Comment:
There is no reason to do the set operations here. Instead, just use `get` on the dictionary for each value in the `content-encoding` header.